java_gen/pre-written/src/main/java/org/projectfloodlight/openflow/types/IPv4Address.java

```diff
@@ -117,6 +117,61 @@ public boolean isMulticast() {
         return ((rawValue >>> 24) & 0xF0) == 0xE0;
     }
 
+    public enum IPAddressType {
+        THIS_NETWORK,
+        PRIVATE_NETWORK,
+        PUBLIC_DATA_NETWORK,
+        CABLE_TV_NETWORK,
+        INTERNET_HOST_LOOPBACK_NETWORK,
+        LINK_LOCAL_NETWORK,
+        TEST_NET,
+        RELAY_ANYCAST_NETWORK,
+        INTERCONNECT_NETWORK,
+        MULTICAST_NETWORK,
+        RESERVED_NETWORK,
+        HOST_ADDRESS
+    }
+
+    public IPAddressType isHostAddress() {
+        if (IPv4AddressWithMask.THIS_NETWORK.contains(this))
+            return IPAddressType.THIS_NETWORK;
+
+        if ((IPv4AddressWithMask.PRIVATE1_NETWORK.contains(this)) ||
```

Comment:
This if-then-else cascade might be expensive if executed on lots of addresses would be better to use a trie.